doc/list.md

```diff
@@ -1 +1,195 @@
 # List
+
+A `List` looks similar to an [Array](Array.md) but it is a pure Lean inductive type that does not
+have an optimized runtime other than what the Lean compiler builds from the List implementation.
+
+```lean
+# namespace hidden
+inductive List (α : Type u) where
+  | nil : List α
+  | cons (head : α) (tail : List α) : List α
+# end hidden
+```
+
+A list then is defined recursively through concatenation of some `head` item with an existing `tail`
+via the `cons` constructor.
+
+## Literals
+
+You can create lists in several ways. You can start with a small list by listing consecutive values between
+`[` and `]` and separated by commas, as shown in the following examples.
+
+```lean
+#check [1, 2, 3] -- List Nat
+
+#check #[1, 2, 3] -- Array Nat - arrays have a # prefix.
+
+#check [] -- List ?m  -- unresolved type.
+#check List.nil (α := String) -- List String - specify the type parameter
+
+#eval List.cons "a" ["b", "c", "d"]     -- ["a", "b", "c", "d"]
+```
+
+Notice that Lean is able to infer the list element type in these cases, createing `List nat` and `List String`.
+
+Notice also that `[]` is short hand for `List.nil` and that it is missing any list element type. To
+resolve this you can write the following:
+
+There is special syntax for `cons`, namely, `::` as follows:
+
+```lean
+#eval 0 :: [1, 2, 3]        -- [0, 1, 2, 3]
+```
+
+The type of the list elements is inferred from the literals used and must be consistent.
+```lean
+#check ["hello", "world"] -- List String
+
+-- The following is not valid, the List element type must be homogeneous
+#check_failure [10, "hello"]
+```
+Recall that the command `#check_failure <term>` only succeeds when the given term is not type correct.
+
+You can also invoke type inference on empty lists using parens to type the expression:
```

Comment:
Type ascription should also be explained in a different part of the manual.

Reply:
Perhaps I found this information particularly missing on lists because there is a concept of empty container.  Non-container types don't really have this problem, except 'Option.none'.